pkg/download/bucket/bucket.go

```diff
@@ -104,25 +105,38 @@ const (
 	updatable   = "updatable"
 )
 
+// bucket holds all values we need to check before we persist the object
+type bucket struct {
+	Name      string `json:"bucketName"`
+	Updatable *bool  `json:"updatable,omitempty"`
+	Status    string `json:"status"`
+}
+
 func convertObject(o []byte, projectName string) (config.Config, error) {
-	r, err := templatetools.NewJSONObject(o)
-	if err != nil {
+	var b bucket
+	if err := json.Unmarshal(o, &b); err != nil {
 		return config.Config{}, fmt.Errorf("failed to unmarshal bucket: %w", err)
 	}
 
-	id, ok := r.Get(bucketName).(string)
-	if !ok {
-		return config.Config{}, fmt.Errorf("variable %q unreadable", bucketName)
+	// bucketName acts as the id, thus it must be set
+	if b.Name == "" {
+		return config.Config{}, fmt.Errorf("bucketName is not set")
 	}
 
 	// skip unmodifiable buckets
-	if u, ok := r.Get(updatable).(bool); ok && !u || buckettools.IsDefault(id) {
-		return config.Config{}, skipErr{fmt.Sprintf("bucket %q is immutable", id)}
+	if b.Updatable != nil && *b.Updatable == false || buckettools.IsDefault(b.Name) {
```

Comment:
If I got it correctly, the field is set not only if a bucket is a default object, but also if it's currently in a state where not changes are allowed, like 'deleting'?

I really think it would then be worth introducing in a general way, and also using if for the check in delete.
There's no reason to try and delete a bucket that's already being deleted.

Reply:
No, updateable is only set if the bucket is a predefined object.

Reply:
Ah! I missed the code on line 132 😅